kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,150 @@
+
+        # Saving list of plots
+        plots_list = []
+        for index in range(5):
+            plots_list.append(plt.figure())
+            plt.plot([1,2,3],[4,5,6], color=colour)
+        list_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplotlib_list"
+        )
+        list_plot_writer.save(plots_list)
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        bucket_name: str,
+        filepath: str,
+        s3fs_args: Optional[Dict] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            bucket_name: Name of the bucket without "s3://" prefix.
+            filepath: Key path to a matplot object file.
+            s3fs_args: Arguments for ``S3FileSystem``. See
+                https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem
+            credentials: Argument for ``client_kwargs``. If needed ``aws_access_key_id``
+                and ``aws_secret_access_key`` are provided here.
+            save_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+
+        _credentials = copy.deepcopy(credentials) or {}
+
+        self._s3fs_args = copy.deepcopy(s3fs_args) or {}
+
+        self._filepath = filepath
+        self._save_args = save_args if save_args else dict()
+        self._bucket_name = bucket_name
+
+        self._s3 = S3FileSystem(client_kwargs=_credentials, **self._s3fs_args)
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            bucket_name=self._bucket_name,
+            filepath=self._filepath,
+            s3fs_args=self._s3fs_args,
+        )
+
+    def _load(self) -> None:
+        raise DataSetError(
+            "Loading not supported for `{}`".format(self.__class__.__name__)
+        )
+
+    def _save(self, data: Union[figure, List[figure], Dict[str, figure]]) -> None:
+        if isinstance(data, list):
+            for index, plot in enumerate(data):
+                key_path = self._filepath + "/" + str(index) + ".png"
+                self._save_to_s3(key_name=key_path, plot=plot)
+
+        elif isinstance(data, dict):
+            for plot_name, plot in data.items():
+                key_path = self._filepath + "/" + plot_name
+                self._save_to_s3(key_name=key_path, plot=plot)
+
+        else:
+            self._save_to_s3(key_name=self._filepath, plot=data)
+
+    def _save_to_s3(self, key_name: str, plot: figure):
+
+        bytes_object = io.BytesIO()
+        plot.savefig(bytes_object, **self._save_args)
+
+        full_key_path = "{}/{}".format(self._bucket_name, key_name)
+
+        with self._s3.open(str(full_key_path), mode="wb") as s3_file:
+            s3_file.write(bytes_object.getvalue())
```

Comment:
Just curious, how would users be able to see the saved plot (in a human readable way, not bytes), since load is not supported for this dataset?

Reply:
You would have to download the plots from s3 to view them.